src/mpl/include/mpl_gpu.h

```diff
@@ -27,4 +27,7 @@ int MPL_gpu_ipc_get_mem_handle(MPL_gpu_ipc_mem_handle_t * h_mem, void *ptr);
 int MPL_gpu_ipc_open_mem_handle(void **ptr, MPL_gpu_ipc_mem_handle_t h_mem);
 int MPL_gpu_ipc_close_mem_handle(void *ptr);
 
+int MPL_gpu_malloc_host(void **ptr, size_t size);
+int MPL_gpu_free_host(void *ptr);
```

Comment:
Would it be better to use "buffer registration", rather than "buffer allocation"?  With buffer registration, we can use this functionality for shared memory too.  With buffer allocation, we'd lose that functionality.

Reply:
A second thought, how about having both `buffer allocation` and `buffer registration`. In the backend of CUDA, they represent `cudaMallocHost` and `cudaHostRegister`.

Reply:
Is there a need for `buffer allocation`?  Is there any benefit for having it compared with `malloc` + `buffer registration`?

Reply:
I see. You just want `malloc` to be pure CPU, and `registration` can be GPU or shared-memory.

Reply:
There's no registration for shared memory.  That needs to be an actual allocation.  So we cannot have GPUs require allocation too.